Bluetooth: Mesh: Increase advertising thread stack size

erwango opened the pull request:
With BT_HOST_CRYPTO, advertising stack size could be overflowed,
increase size to 1024 when BT_HOST_CRYPTO is enabled.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

jhedberg commented on subsys/bluetooth/host/mesh/adv.c:
```diff
@@ -45,7 +45,11 @@
 #define ADV_INT_DEFAULT_MS 100
 #define ADV_INT_FAST_MS    20
 
+#ifdef CONFIG_BT_HOST_CRYPTO
```
minor thing: we've tried to be consistent with `#if defined(CONFIG_...)` in the Bluetooth code, could you please do that? (benefit is that you can combine multiple `defined(...)` conditions for a single `#if` but not with `#ifdef`)

erwango committed:
Bluetooth: Mesh: Increase advertising thread stack size

With BT_HOST_CRYPTO, advertising stack size could be overflowed,
increase size to 1024 when BT_HOST_CRYPTO is enabled.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

jhedberg commented:
@sjanc @Vudentz @carlescufi could we get a second review on this? A month ago or so it was discovered that the stack requirement for advertising had grown, see commit 1cfa12276c3723f0b84401454e65ba2e5956828b, however turns out we still need the conditionally higher size for BT_HOST_CRYPTO.

carlescufi reviewed (approved):
No objections
